pkg/network/usm/monitor_test.go

```diff
@@ -845,11 +842,21 @@ func (s *USMHTTP2Suite) TestHTTP2KernelTelemetry() {
 
 			tt.runClients(t, 1)
 
+			// We cannot predict if the client will send an RST frame or not, thus we cannot predict the number of
+			// frames with EOS or RST frames, which leads into a flaking test. Therefore, we are asserting that the
+			// gotten number of EOS or RST frames is at least the number of expected EOS frames.
+			expectedEOSOrRST := tt.expectedTelemetry.End_of_stream + tt.expectedTelemetry.End_of_stream_rst
+			tt.expectedTelemetry.End_of_stream = 0
+			tt.expectedTelemetry.End_of_stream_rst = 0
```

Comment:
I understand that you did it for the deep equal comparison, right? So I am wondering if it might make more sense to use the exact fields in the kernel metrics struct. You already introduced a nice solution for the EOS and RST, so maybe we can compare only the remaining three fields. This way, we wouldn't need to add this part for the expected telemetry and the telemetry we received. WDYT?

Reply:
If I understand correctly your suggestion, you mean to replace our DeepEqual with:

```go
return telemetry.Request_seen == tt.expectedTelemetry.Request_seen &&
	telemetry.Response_seen == tt.expectedTelemetry.Response_seen &&
	telemetry.Path_exceeds_frame == tt.expectedTelemetry.Path_exceeds_frame &&
	telemetry.Exceeding_max_interesting_frames == tt.expectedTelemetry.Exceeding_max_interesting_frames &&
	telemetry.Exceeding_max_frames_to_filter == tt.expectedTelemetry.Exceeding_max_frames_to_filter &&
	sum >= expectedEOSOrRST &&
	reflect.DeepEqual(telemetry.Path_size_bucket, tt.expectedTelemetry.Path_size_bucket)
```

Reply:
yes, Personally I think it is more clear to use it like that